examples/twittermap/web/public/javascripts/common/module-manager.js

```diff
@@ -0,0 +1,81 @@
+angular.module("cloudberry.common")
+  .service("moduleManager", function() {
+
+    var moduleManager = {
+
+      // Built-in events
+      EVENT: {
+        CLICK_SEARCH_BUTTON: "clickSearchButtonEvent",
+        CHANGE_TIME_SERIES_RANGE: "changeTimeSeriesRangeEvent",
+        CHANGE_ZOOM_LEVEL: "changeZoomLevelEvent",
+        CHANGE_REGION_BY_DRAG: "changeRegionByDragEvent",
+        CHANGE_MAP_TYPE: "changeMapTypeEvent"
+      },
+
+      eventsListeners: {},
+
+      /**
+       * subscribe an event:
+       *   once this event happens, the eventHandler function will be called.
+       *
+       * @param eventName moduleManager.EVENT, event name
+       * @param eventHandler function, callback function(event): event is the content object of the event
+       * @returns {boolean} true: if subscribed successfully, false: otherwise
+       */
+      subscribeEvent(eventName, eventHandler) {
+
+        if (!(eventName in this.EVENT)) {
+          return false;
+        }
+
+        if (eventHandler instanceof Function) {
+          if (eventName in this.eventsListeners) {
+            this.eventsListeners[eventName].add(eventHandler);
+          }
+          else {
+            this.eventsListeners[eventName] = new Set([eventHandler]);
```

Comment:
Is [eventHandler] correct? Should it be Set(eventHandler)?

Reply:
This is correct. https://www.geeksforgeeks.org/sets-in-javascript/
The parameter for construction of Set must be iterable object instead of single object.

Reply:
OK.